### PR TITLE
Change mirror write  retry to run after mirror scan and not in parallel

### PR DESCRIPTION
**Pull request by liranmauda**
### Explain the Changes
- Change mirror write retry to run after mirror scan and not in parallel to avoid false positives in the test. In some cases, the test failed with:
```
   1) mirror_writer
        run_batch
          should get a retry scanner with the failed range after failure:

       AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:

   assert(get_builder.secondCall.calledWithExactly({

       + expected - actual

       -false
       +true

       at Context.<anonymous> (src/test/unit_tests/internal/test_mirror_writer.js:131:13)
```

- Converted the test to jest
- Added some test cases to increase coverage

## Summary by CodeRabbit

* **Chores**
  * More robust background mirror-writing: improved error detection/reporting, refined retry behavior, and clearer end-of-batch delay selection for steadier processing.
  * Improved metadata chunk iteration: async paging with precise range semantics and reliable end-of-range signaling for consistent resume behavior.

* **Tests**
  * Replaced and expanded unit tests for mirror-writing batch, retry, marker handling, and edge cases; removed an older redundant test registration.

### CODE REVIEW

File: src/server/bg_services/mirror_writer.js [hunk 1]
```diff
@@ -52,25 +52,31 @@ class MirrorWriter {
...
-            const [scan_res, retry_res] = [await this.mirror_scanner.run_batch(bucket_ids), await this._retry_mirror(bucket_ids)];
+            const scan_res = await this.mirror_scanner.run_batch(bucket_ids);
+
+            // push errors to errors_marker before retry so _retry_mirror can create retry_scanner with the failed range in same batch
+            if (!scan_res.successful && scan_res.chunk_ids && scan_res.chunk_ids.length) {
+                this._handle_build_errors(scan_res.chunk_ids);
+            }
+
+            // run retry after pushing scan errors above so _retry_mirror can create retry_scanner with the failed range in this batch
+            const retry_res = await this._retry_mirror(bucket_ids);
```
Comment (dannyzaken) [inline src/server/bg_services/mirror_writer.js:61]: Shouldn't we run the retry in the next batch and not right away?

Comment (liranmauda) [reply src/server/bg_services/mirror_writer.js:61]: We do that so that if the scanner fails will run on the same batch.
On the next batch, we will run the failed on the retry.

File: src/server/bg_services/mirror_writer.js [hunk 2]
```diff
@@ -50,58 +50,38 @@ class MirrorWriter {
...
-
         const bucket_ids = mirrored_buckets.map(bucket => bucket._id);
 
         try {
-            const [scan_res, retry_res] = [await this.mirror_scanner.run_batch(bucket_ids), await this._retry_mirror(bucket_ids)];
+            const scan_res = await this.mirror_scanner.run_batch(bucket_ids);
+            // Push scan errors before retry so _retry_mirror can create retry_scanner with the failed range in this batch
+            if (!scan_res.successful && scan_res.chunk_ids?.length) {
+                this._handle_build_errors(scan_res.chunk_ids);
+            }
```
[the remark this answers is not in the document]
Comment (liranmauda) [reply src/server/bg_services/mirror_writer.js:60]: @coderabbitai Fixed in md_store.js, verify me

**Review by dannyzaken — APPROVED** [review]
